### PR TITLE
feat: Adding KTX for Maps SDK V3 Beta

### LINKED ISSUE #48: Support KTX for V3 Beta
**Is your feature request related to a problem? Please describe.**
The KTX library at the moment is only for the Google Play Services SDK and not the standalone V3 beta. The standalone V3 library should also be supported.

 **Describe the solution you'd like**
Add an additional module called `maps-v3-ktx` that uses Maps V3 beta instead of Maps on Play Services.

 **Describe alternatives you've considered**
N/A

 **Additional context**
N/A

### PR DESCRIPTION
**Pull request by arriolac**
Pretty much what's in `maps-ktx`.

Closes #48

### CODE REVIEW

**Comment by arriolac** [comment]
@barbeau I ended up changing my approach from using product flavors to autogenerating autogenerating source code into a new module (`maps-v3-ktx`) instead. The reason I created a new module is due to challenges faced related to generating artifacts and also with modifying the publishing workflow. 

### Challenges
* the `-sources.jar` file was empty for both gms and standalone product flavors (this is because the sources jar generates its code from within the `/main` source set. It looks like the `sourcesJar` Gradle task is hardcoded to that directory and cannot be modified to point to another dir per flavor. This is perhaps an improvement that can be made within the `com.android.library` gradle plugin.
* the generated `-javadoc.jar` file only created docs for one flavor and not the other. I encountered similar issues w/ pointing to a different dir per flavor.
* the current publishing tasks in `./build.gradle` assumes one publishable artifact per project. With the addition of flavors, the code became much more complex and harder to reason about.

### Future work

As currently implemented, the code within `maps-v3-ktx` is not meant to be editable as mentioned in the module README. It would be nice for this to be editable though since the SDK within Play Services and the V3 BETA can and will diverge (i.e. features will arrive sooner within BETA and eventually land in Play Services). I think this is doable by moving source code generation to a separate script that only runs when there are changes within `maps-v3` and can either handle merging or let you know when there are changes in the existing file so that it can be merged manually. Won't implement this here but thought I should note.

File: maps-ktx/build.gradle [hunk 1]
```diff
@@ -31,10 +31,48 @@ android {
     }
 }
 
+/**
+ * This task will copy all of the source code in this module into the `maps-v3-ktx` module
+ * and change its imports from the Maps SDK on Play Services to the Standalone V3 Beta Maps SDK.
+ */
+task copyGmsToV3BetaSource {
+    inputs.dir 'src/main/java/com/google/maps/android/ktx'
+    outputs.dir '../src/main/java/com/google/maps/android/ktx'
```
Comment (barbeau) [inline maps-ktx/build.gradle:40]: I believe this `outputs.dir` should be:

```
'../maps-v3-ktx/src/main/java/com/google/maps/android/ktx'
```

File: maps-ktx/build.gradle [hunk 2]
```diff
@@ -31,10 +31,48 @@ android {
...
+    }
+}
+
+/**
+ * This task will copy all of the tests in this module into the `maps-v3-ktx` module
+ * and change its imports from the Maps SDK on Play Services to the Standalone V3 Beta Maps SDK.
+ */
+task copyGmsToV3BetaTest {
+    inputs.dir 'src/test/java/com/google/maps/android/ktx'
+    outputs.dir '../src/main/java/com/google/maps/android/ktx'
```
Comment (barbeau) [inline maps-ktx/build.gradle:58]: Similar here - I believe this `outputs.dir` should be:

```
'../maps-v3-ktx/src/test/java/com/google/maps/android/ktx'
```

File: maps-v3-ktx/README.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+maps-v3-ktx
+===========
+
+This module contains Kotlin extensions (KTX) for the Maps SDK V3 BETA library for Android. 
+
+**NOTE**: The source code in this module is autogenerated from source code within the `maps-ktx` 
+and is not intended for modifications. If any edits should be made, they should be made in that 
+module first.
```
Comment (barbeau) [inline maps-v3-ktx/README.md:8]: I would add when/how the code is generated:

```suggestion
**NOTE**: The source code in this module is autogenerated from source code within the `maps-ktx` 
module (on `maps-ktx` compilation) and is not intended for modifications. Please make any code 
changes in the `maps-ktx` module.
```

Comment (arriolac) [reply maps-v3-ktx/README.md:8]: Good call 💯

**Pull request merged**